pyins/tests/test_transform.py

```diff
@@ -16,12 +20,93 @@ def test_lla_to_ecef():
     assert_allclose(r_e, [[earth.A + 10, 0, 0], [0, 0, -b + 10]], atol=1e-9)
 
 
+def test_lla_to_ned():
+    lla = [[0, 0, 1000], [90, 90, 0], [0, -90, -1000]]
+    ned = transform.lla_to_ned(lla)
+    a = earth.A + 1000
+    b = (1 - earth.E2) ** 0.5 * earth.A
+    test = [[0, 0, 0], [b, 0, a], [0, -earth.A + 1000, a]]
+    assert_allclose(ned, test, atol=1e-8)
+    assert(isinstance(ned, np.ndarray))
+
+    lla = pd.DataFrame(data=lla, columns=LLA_COLS)
+    ned = transform.lla_to_ned(lla)
+    test = pd.DataFrame(data=test, columns=NED_COLS)
+    assert_allclose(ned, test, atol=1e-8)
+    assert(isinstance(ned, pd.DataFrame))
+
+    lla = [[0, 0, 1000], [90, 90, 0], [0, -90, -1000]]
+    ned = transform.lla_to_ned(lla, [0, -90, -1000])
+    a = earth.A - 1000
+    test = [[0, earth.A + 1000, a], [b, 0, a], [0, 0, 0]]
+    assert_allclose(ned, test, atol=1e-8)
+    assert(isinstance(ned, np.ndarray))
+
+    lla = pd.DataFrame(data=lla, columns=LLA_COLS)
+    ned = transform.lla_to_ned(lla, [0, -90, -1000])
+    test = pd.DataFrame(data=test, columns=NED_COLS)
+    assert_allclose(ned, test, atol=1e-8)
+    assert(isinstance(ned, pd.DataFrame))
+
+
 def test_perturb_ll():
     lla = [40, 50, 0]
     lla_new = transform.perturb_lla(lla, [10, -20, 5])
     lla_back = transform.perturb_lla(lla_new, [-10, 20, -5])
     assert_allclose(lla_back, lla, rtol=1e-11)
 
+    lla = [[40, 50, 0], [-40, 50, 10]]
+    lla_new = transform.perturb_lla(lla, [10, -20, 5])
+    lla_back = transform.perturb_lla(lla_new, [-10, 20, -5])
+    assert_allclose(lla_back, lla, rtol=1e-11)
+
+
+def test_translate_trajectory():
+    traj, imu = sinusoid_velocity_motion(0.1, 60, [10, 20, -3], [5, 7, 3])
+    traj_new = transform.translate_trajectory(traj, [10, -20, 5])
+    mat = Rotation.from_euler('xyz', traj[RPH_COLS], degrees=True).as_matrix()
+    ned = util.mv_prod(mat, [10, -20, 5])
+    lla_test = transform.perturb_lla(traj[LLA_COLS], ned)
+    assert_allclose(traj_new[LLA_COLS], lla_test, rtol=1e-11)
+    assert_allclose(traj_new[VEL_COLS], traj[VEL_COLS], rtol=1e-11)
+    assert_allclose(traj_new[RPH_COLS], traj[RPH_COLS], rtol=1e-11)
+
+    traj[RATE_COLS] = imu[GYRO_COLS].values
```

Comment:
What if we generate a trajectory using shifted lla (with `sim.generate_imu`) then velocity should be properly adjusted and we can compare it with velocity adjusted by cross-product. 

I think it would be an interesting and useful test.

Reply:
good idea!  but lets do it in seperate MR, with  better implementation of smooth_state function, etc